sys/include/ecc/hamming256.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2015 Lucas Jenß
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_ecc
+ * @brief
+ * @{
+ *
+ * @brief       Hamming Code implementation for 256byte data segments
+ * @author      Lucas Jenß <lucas@x3ro.de>
+ */
+
+#ifndef _HAMMING256_H
+#define _HAMMING256_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**  No bit errors were detected in the message */
+#define Hamming_ERROR_NONE 0
+
+/**  A single bit was incorrect but has been recovered. */
+#define Hamming_ERROR_SINGLEBIT         1
+
+/** The original code has been corrupted. */
+#define Hamming_ERROR_ECC               2
+
+/** Multiple bits are incorrect in the data and they cannot be corrected. */
+#define Hamming_ERROR_MULTIPLEBITS      3
+
+
+/**
+ *  @brief Computes 3-bytes hamming codes for a data block whose size is multiple of
+ *  256 bytes. Each 256 bytes block gets its own code.
```

Comment:
According to doxygen brief automatically ends at the first sentence and everything afterwards is put into the detailed docs. Looked kind of weird having the line on its own. Will change.